File: movie-review-system/frontend/src/types/movie.ts
export interface Movie {
  movie_id: number;
  tmdb_id: number | null;
  title: string;
  original_title: string | null;
  description: string | null;
  genres: string[] | null;
  release_year: number | null;
  poster_url: string | null;
  external_rating: number | null;
}

export interface MovieSummary {
  summary_id: number | null;
  movie_id: number;
  average_user_rating: number | null;
  average_predicted_rating: number | null;
  review_count: number;
  sentiment_distribution: Record<string, number> | null;
  aspect_scores: Record<string, number> | null;
  aspect_frequency: Record<string, number> | null;
  updated_at: string | null;
}

File: movie-review-system/backend/app/schemas/movie.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class MovieBase(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    original_title: str | None = Field(default=None, max_length=500)
    description: str | None = None
    genres: list[str] | None = None
    release_year: int | None = Field(default=None, ge=1888, le=2100)
    poster_url: str | None = None
    external_rating: float | None = Field(default=None, ge=0, le=10)


class MovieCreate(MovieBase):
    tmdb_id: int | None = None


class MovieListItem(MovieBase):
    movie_id: int
    tmdb_id: int | None = None

    model_config = ConfigDict(from_attributes=True)


class MovieRead(MovieListItem):
    pass


class MovieSummaryRead(BaseModel):
    summary_id: int | None = None
    movie_id: int
    average_user_rating: float | None = None
    average_predicted_rating: float | None = None
    review_count: int = 0
    sentiment_distribution: dict[str, int] | None = None
    aspect_scores: dict[str, float] | None = None
    aspect_frequency: dict[str, int] | None = None
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)
